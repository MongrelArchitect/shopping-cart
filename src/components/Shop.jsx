import React, { useState, useEffect } from 'react';
import Product from './Product';

export default function Shop() {
  const [products, setProducts] = useState(null);

  useEffect(() => {
    const fetchData = async () => {
      const response = await fetch('https://fakestoreapi.com/products');
      const result = await response.json();
      setProducts(result);
    };

    fetchData();
  }, []);

  if (products) {
    return (
      <div className="products">
        {products.map((product) => (
          <Product product={product} key={product.id} />
        ))}
      </div>
    );
  }

  return (
    <div>
      <span>Loading...</span>
    </div>
  );
}
