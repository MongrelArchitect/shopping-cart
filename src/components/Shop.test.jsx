import React from 'react';
import { render } from '@testing-library/react';
import Shop from './Shop';

jest.mock(
  './Product',
  () => function MockProduct({ product }) {
    return (
      <div className="product">
        <span className="product-title">{product.title}</span>
        <img src={product.image} alt="" />
        <span>{product.description}</span>
        <span className="product-price">
          $
          {product.price}
        </span>
        <label htmlFor={`quant${product.id}`}>
          Quantity:
          <input type="number" min="1" id={`quant${product.id}`} />
        </label>
        <button type="button">ADD TO CART</button>
      </div>
    );
  },
);

describe('Shop component', () => {
  it('renders correctly with zero products', () => {
    const { container } = render(
      <Shop products={null} addToCart={jest.fn()} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders correctly with some products', () => {
    const mockProducts = [
      {
        title: 'Fake Product #1',
        image: '#',
        description: 'A fantastic fake product that does nothing',
        id: 1,
        price: 200,
      },
      {
        title: 'Fake Product #2',
        image: '#',
        description: 'Another phony placeholder',
        id: 2,
        price: 25,
      },
    ];

    const { container } = render(
      <Shop products={mockProducts} addToCart={jest.fn()} />,
    );
    expect(container).toMatchSnapshot();
  });
});
