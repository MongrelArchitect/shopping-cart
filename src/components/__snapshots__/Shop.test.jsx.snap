// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shop component renders correctly with some products 1`] = `
<div>
  <div
    class="products"
  >
    <div
      class="product"
    >
      <span
        class="product-title"
      >
        Fake Product #1
      </span>
      <img
        alt=""
        src="#"
      />
      <span>
        A fantastic fake product that does nothing
      </span>
      <span
        class="product-price"
      >
        $
        200
      </span>
      <label
        for="quant1"
      >
        Quantity:
        <input
          id="quant1"
          min="1"
          type="number"
        />
      </label>
      <button
        type="button"
      >
        ADD TO CART
      </button>
    </div>
    <div
      class="product"
    >
      <span
        class="product-title"
      >
        Fake Product #2
      </span>
      <img
        alt=""
        src="#"
      />
      <span>
        Another phony placeholder
      </span>
      <span
        class="product-price"
      >
        $
        25
      </span>
      <label
        for="quant2"
      >
        Quantity:
        <input
          id="quant2"
          min="1"
          type="number"
        />
      </label>
      <button
        type="button"
      >
        ADD TO CART
      </button>
    </div>
  </div>
</div>
`;

exports[`Shop component renders correctly with zero products 1`] = `
<div>
  <div
    class="cart"
  >
    <h3>
      Loading...
    </h3>
  </div>
</div>
`;
